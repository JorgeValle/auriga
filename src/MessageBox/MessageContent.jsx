import React from 'react';

// Components should represent one piece of data model
class MessageContent extends React.Component {

  constructor(props) {

    super(props);
    this.handleMessageDown = this.handleMessageDown.bind(this);
    this.handleMessageUp = this.handleMessageUp.bind(this);
  }

  handleMessageDown(e) {
    this.props.onMessageDown(e.target);
  }

  handleMessageUp(e) {
    console.log('e.target is:');
    console.log(e.target);
    this.props.onMessageUp(e.target);
  }

  render() {

    const messages = this.props.messages;

    const messageNum = this.props.messageNum;

    console.log('Props on MessageContent component');
    console.log(this.props);

    console.log('State on MessageContent component');
    console.log(this.state);

    return (
      <div>
        <p className="message-content">{messages[messageNum]}</p>
        <nav className="message-nav">
          <button onClick={this.handleMessageDown}>First</button>
          <button onClick={this.handleMessageDown}>Previous</button>
          <button onClick={this.handleMessageUp}>Next</button>
          <button onClick={this.handleMessageUp}>Latest</button>
        </nav>
        <span>{messageNum} / {messages.length}</span>

      </div>

    );
  }
}

export default MessageContent;