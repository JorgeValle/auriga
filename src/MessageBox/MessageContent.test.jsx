import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import { shallow, configure } from 'enzyme';
import MessageContent from './MessageContent.jsx';

configure({
  adapter: new Adapter()
});

describe('<MessageContent />', () => {

  // We shallow render to make sure changes in children don't cause cascading fails
  it('shallow renders', () => {
    const wrapper = shallow(<MessageContent />);
  });

  // Snapshot matching
  it('matches snapshot', () => {
    const wrapper = shallow(<MessageContent />);
    expect(wrapper).toMatchSnapshot();
  });

});