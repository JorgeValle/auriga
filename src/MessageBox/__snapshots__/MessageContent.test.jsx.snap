// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MessageContent /> matches snapshot 1`] = `ShallowWrapper {}`;
